helmctlr/helm.go

```diff
@@ -175,7 +175,7 @@ func (c Controller) releaseExists(rlsName string) bool {
 		helm.ReleaseListFilter(rlsName),
 		helm.ReleaseListStatuses(statuses),
 	)
-	if err != nil {
+	if err != nil || r == nil {
```

Comment:
I'm very interested in how r is nil. According to https://godoc.org/k8s.io/helm/pkg/helm#Client.ListReleases the response is a ListReleasesResponse, which means r should not be nil, even if there are no releases. I think something else is happening, because if you start lostromos with a new cluster and then add the CRD everything does work. Which is also how the entire integration test suite runs. 

Can you provide steps to replicate this?

Reply:
if https://github.com/kubernetes/helm/blob/master/pkg/helm/client.go#L353 is true, resp is still nil which could result into this.

Reply:
received a panic while testing in k8s deployment in dev, reason for making this fix. There is a possibility for this to fail, irrespective of how it fails.

Reply:
ERROR: logging before flag.Parse: E0530 17:59:20.033627       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/home/travis/gopath/src/github.com/wpengine/lostromos/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/home/travis/gopath/src/github.com/wpengine/lostromos/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/home/travis/gopath/src/github.com/wpengine/lostromos/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/home/travis/.gimme/versions/go1.9.5.linux.amd64/src/runtime/asm_amd64.s:509
/home/travis/.gimme/versions/go1.9.5.linux.amd64/src/runtime/panic.go:491
/home/travis/.gimme/versions/go1.9.5.linux.amd64/src/runtime/panic.go:63
/home/travis/.gimme/versions/go1.9.5.linux.amd64/src/runtime/signal_unix.go:367
/home/travis/gopath/src/github.com/wpengine/lostromos/helmctlr/helm.go:181
/home/travis/gopath/src/github.com/wpengine/lostromos/helmctlr/helm.go:133
/home/travis/gopath/src/github.com/wpengine/lostromos/helmctlr/helm.go:69
<autogenerated>:1
/home/travis/gopath/src/github.com/wpengine/lostromos/crwatcher/watcher.go:119

Reply:
I'll approve and merge, but even on line 353 it would return an empty `ListReleasesResponse` not a nil. So I would like to know more about how this is happening, because I have some doubts that this is what the underlying problem is.

Reply:
ln 349, resp is a pointer, reason for it to be nil, when not allocated.